torchvision/datasets/kinetics.py

```diff
@@ -0,0 +1,26 @@
+from .video_utils import VideoClips
+from .utils import list_dir
+from .folder import make_dataset
+from .vision import VisionDataset
+
+
+class KineticsVideo(VisionDataset):
+    def __init__(self, root, frames_per_clip, step_between_clips=1):
+        super(KineticsVideo, self).__init__(root)
+        extensions = ('avi',)
+
+        classes = list(sorted(list_dir(root)))
+        class_to_idx = {classes[i]: i for i in range(len(classes))}
+        self.samples = make_dataset(self.root, class_to_idx, extensions, is_valid_file=None)
+        self.classes = classes
+        video_list = [x[0] for x in self.samples]
+        self.video_clips = VideoClips(video_list, frames_per_clip, step_between_clips)
+
+    def __len__(self):
+        return self.video_clips.num_clips()
+
+    def __getitem__(self, idx):
```

Comment:
The only concern I have about this is the fact that we sample all clips all the time. In principle, I think we'd want to be able to sample 5 randomly selected clips per video at every epoch. Should we leave that as an exercise to the user, as it's fairly easy to do, or should we handle this somehow within the dataset?

Reply:
I'll have to handle this when I implement the training code for the video models, thanks for the reminder!

So, to double check, in order to get the best results, you randomly sample 5 clips per video during training, and resamples that every epoch, right?

Reply:
No worries;
yes, for kinetics, we obtained the best result by sampling 5 clips from each video at every epoch such that starting point for every clip is chosen at random.

Reply:
I've addressed this with the latest version via a custom `Sampler`.

Let me know what you think